File: client/src/model/GenericPage.ts
export interface DefaultPagination {
  page: number;
  size: number;
  last: boolean;
  totalElements: number;
  totalPages: number;
}

export interface GenericPage<T> extends DefaultPagination {
  content: T[];
}

export const DEFAULT_PAGINATION: DefaultPagination = Object.freeze({
  page: 0,
  size: 20,
  last: true,
  totalElements: 0,
  totalPages: 0,
});

File: server/src/main/java/com/alexandrecampos/devchallenge/dto/GenericPage.java
package com.alexandrecampos.devchallenge.dto;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class GenericPage<T> {
    private List<T> content;
    private int page;
    private int size;
    private boolean last;
    private int totalElements;
    private int totalPages;
}
